core/src/main/java/org/apache/calcite/sql2rel/ConvertToChecked.java

```diff
@@ -50,13 +51,18 @@ public ConvertToChecked(RexBuilder builder) {
    * Visitor which rewrites an expression tree such that all
    * arithmetic operations that produce numeric values use checked arithmetic.
    */
-  static class ConvertRexToChecked extends RexShuttle {
+  class ConvertRexToChecked extends RexShuttle {
     private final RexBuilder builder;
 
     ConvertRexToChecked(RexBuilder builder) {
       this.builder = builder;
     }
 
+    @Override public RexNode visitSubQuery(RexSubQuery subQuery) {
+      RelNode result = subQuery.rel.accept(ConvertToChecked.this);
```

Comment:
SubQuery has operands that need to handle? maybe SQL like:
```
select count(*) as c
from "scott".emp as e
where  -CAST(-32768 AS SMALLINT) not in (
  select deptno
  from dept
  where dname = e.ename)
```
`-CAST(-32768 AS SMALLINT)` will be the Subquery's operands.

Reply:
yes, this calling the visitor recursively on the subquery. This was the bug fixed in this PR: the recursive call wasn't made.

Reply:
Can we add an SQL with a subquery to test this change?

Reply:
The test in the issue fails without this change: `SELECT -CAST(-32768 AS SMALLINT)`
Turns out that because of the way Prepare works, this is a RexSubquery at the point where this visitor is invoked.

Reply:
Ok, I test it in local mode, This SQL generates an SCALAR_QUERY in RelNode(It's strange for this SQL to generate SCALAR_QUERY, but it has nothing to do with this PR). So the test is valid.